src/autocomplete/QueryMatcher.ts

```diff
@@ -75,7 +75,11 @@ export default class QueryMatcher<T> {
         this._items = new Map();
 
         for (const object of objects) {
-            const keyValues = _at(object, this._keys);
+            // Need to use unsafe coerce here because the objects can have any
+            // type for their values. We assume that those values who's keys have
+            // been specified will be string. Also, we cannot infer all the
+            // types of the keys of the objects at compile.
+            const keyValues: (string)[] = _at<T>(object as any, this._keys) as any;
```

Comment:
`_at` is actually looking for a `string` type, not `T`. Also the array type isn't needed when specifying the type to the `_at` function:

```suggestion
            const keyValues = _at<string>(<any>object, this._keys);
```

Reply:
Nice, much cleaner